Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs

```diff
@@ -359,804 +382,439 @@ public TaskMailRepository(IRepository<TaskMail> taskMail, IStringConstantReposit
+                        taskMailReportAcList.Add(taskMailReportAc);
                     }
-
                 }
             }
-            return taskMailAc;
+            return taskMailReportAcList;
         }
 
 
         /// <summary>
         /// TaskMailDetails Information For the selected date
         /// </summary>
-        /// <param name="UserId"></param>
-        /// <param name="UserName"></param>
-        /// <param name="UserRole"></param>
-        /// <param name="CreatedOn"></param>
-        /// <param name="LoginId"></param>
-        /// <param name="SelectedDate"></param>
+        /// <param name="userId"></param>
+        /// <param name="userName"></param>
+        /// <param name="role"></param>
+        /// <param name="createdOn"></param>
+        /// <param name="loginId"></param>
+        /// <param name="selectedDate"></param>
         /// <returns></returns>
-        public async Task<List<TaskMailUserAc>> TaskMailDetailsReportInformationForSelectedDate(string UserId, string UserName, string UserRole, string CreatedOn, string LoginId, string SelectedDate)
+        private async Task<List<TaskMailReportAc>> TaskMailDetailsForSelectedDateAsync(string userId, string userName, string role, string createdOn, string loginId, string selectedDate)
         {
-            List<TaskMailUserAc> taskMailAc = new List<TaskMailUserAc>();
-            List<TaskMailReportAc> taskMailReportAc = new List<TaskMailReportAc>();
-            DateTime? maxDateSelectedUser = null;
-            DateTime? minDateSelectedUser = null;
-
-            DateTime slectedDateForAdmin = Convert.ToDateTime(SelectedDate).Date;
-            var taskMails = await _taskMail.FetchAsync(y => y.EmployeeId == UserId && DbFunctions.TruncateTime(y.CreatedOn) == DbFunctions.TruncateTime(slectedDateForAdmin));
+            List<TaskMailReportAc> taskMailReportAcList = new List<TaskMailReportAc>();
+            List<TaskMailDetailReportAc> taskMailDetailReportAcList = new List<TaskMailDetailReportAc>();
+            DateTime pickDate = Convert.ToDateTime(selectedDate).Date;
+            var taskMails = await _taskMail.FetchAsync(y => y.EmployeeId == userId && DbFunctions.TruncateTime(y.CreatedOn) == DbFunctions.TruncateTime(pickDate));
             if (taskMails.Count() != 0)
             {
-                var maxRecord = await _taskMail.FetchAsync(x => x.EmployeeId == UserId);
-                var maxDate = maxRecord.OrderByDescending(x => x.CreatedOn).FirstOrDefault();
-                maxDateSelectedUser = maxDate.CreatedOn;
-                var minRecord = await _taskMail.FetchAsync(x => x.EmployeeId == UserId);
-                var minDate = minRecord.OrderBy(x => x.CreatedOn).FirstOrDefault();
-                minDateSelectedUser = minDate.CreatedOn;
+                DateTime? maxDate = null;
```

Comment:
I did not get this comment.
I know DateTime? maxDate itself contain null but if we have to check it in"if condition" then we have to assign it as null.
Please correct me if i am wrong